include/deal.II/base/polynomial_space.h

```diff
@@ -270,37 +262,29 @@ PolynomialSpace<3>::compute_index(const unsigned int n) const;
 template <int dim>
 template <class Pol>
 PolynomialSpace<dim>::PolynomialSpace(const std::vector<Pol> &pols)
-  : polynomials(pols.begin(), pols.end())
-  , n_pols(n_polynomials(polynomials.size()))
-  , index_map(n_pols)
-  , index_map_inverse(n_pols)
+  : ScalarPolynomialsBase<dim>(pols.size(), n_polynomials(pols.size()))
+  , polynomials(pols.begin(), pols.end())
+  , index_map(this->n())
+  , index_map_inverse(this->n())
```

Comment:
Is `this->n()` a `virtual` function? If so, this will not work.

Reply:
If it is not a virtual function, then it might be useful to make that clear by writing the call as `this->ScalarPolynomialsBase<dim>::n()`.

Reply:
What about replacing `this->n()` with `n_polynomials(pols.size())` there instead?